File: 2037-minimum-number-of-moves-to-seat-everyone/2037-minimum-number-of-moves-to-seat-everyone.py
class Solution:
    def minMovesToSeat(self, seats: List[int], students: List[int]) -> int:
        n = len(seats)
        position_seat = [0] * 101
        position_stud = [0] * 101

        for pos in seats:
            position_seat[pos] += 1
        for pos in students:
            position_stud[pos] += 1
        i , j = 0 , 0
        moves = 0
        while i <= 100 and j <= 100:
            if position_seat[i] == 0:
                i += 1
            if position_stud[j] == 0:
                j += 1
            if i <= 100 and j <= 100 and position_seat[i] != 0 and position_stud[j] != 0:
                moves += abs(i - j)
                position_seat[i] -= 1
                position_stud[j] -= 1
        return moves



# class Solution:
#     def minMovesToSeat(self, seats: List[int], students: List[int]) -> int:
#         seats.sort()
#         students.sort()
#         n = len(seats)
#         moves = 0
#         for i in range(n):
#             moves += abs(students[i] - seats[i])
#         return moves

File: LeetCode/2037-minimum-number-of-moves-to-seat-everyone/2037-minimum-number-of-moves-to-seat-everyone.cpp
class Solution {
public:
    int minMovesToSeat(vector<int>& seats, vector<int>& students) {
        int n = seats.size();
        int pos_seat[101];
        int pos_stud[101];
        for (int seat : seats) {
            pos_seat[seat]++;
        }
        for (int student : students) {
            pos_stud[student]++;
        }
        int i = 0, j = 0, moves = 0;
        while (i <= 100 && j <= 100) {
            if (pos_seat[i] == 0) {
                i++;
            }
            if (pos_stud[j] == 0) {
                j++;
            }
            if (i <= 100 && j <= 100 && pos_seat[i] != 0 && pos_stud[j] != 0) {
                moves += abs(i - j);
                pos_seat[i]--;
                pos_stud[j]--;
            }
        }
        return moves;
    }
};



// class Solution {
// public:
//     int minMovesToSeat(vector<int>& seats, vector<int>& students) {
//         sort(seats.begin(), seats.end());
//         sort(students.begin(), students.end());
//         int moves = 0;
//         int n = seats.size();
//         for (int i = 0; i < n; i++) {
//             moves += abs(seats[i] - students[i]);
//         }
//         return moves;
//     }
// };